Automattic/jetpack
# Revert "Show social review prompt with both Social and Jetpack installed"

Reverts Automattic/jetpack#29910

We are going to revert this PR to fix the initial state of the plugin p1681469685034749-slack-C02JJ910CNL and pick up 1198218726984184-as-1204393860935466 in the following sprints. The social plugin will have the review prompts working when it's active. The review prompts won't show up when both Social and Jetpack are active. 

This is the https://github.com/Automattic/jetpack/pull/27917 which originally added the review prompts. But this [line](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/social/src/class-jetpack-social.php#L295) sets the state for the Social plugin only if the Jetpack plugin is not active which leads to https://github.com/Automattic/jetpack/pull/29910 being created, I presume. That accidentally stomped over the plugin's initial state.

## Does this pull request change what data or activity we track or use?


* Yes, this PR reinstates 3 new analytics events around the new review prompt:

- jetpack_social_plugin_review_prompt_view (recorded when review prompt is rendered)
- jetpack_social_plugin_review_prompt_new_review_click (recorded when "Leave a review" is clicked)
- jetpack_social_plugin_review_prompt_dismiss_click (recorded when "Dismissed" is clicked)

## Testing instructions:

- When the prompt renders, the jetpack_social_plugin_review_prompt_view analytics event should fire. Search for this in your browser's network panel to confirm.
- Confirm that clicking on "Leave a Review" takes you to the reviews page for the Jetpack Social plugin
- Create another new post, but disable sharing. No prompt will show up since sharing was disabled. 
- Publish a post again, but disable the individual connections. Make sure the prompts don't show up. 
- Publish a post again with sharing enabled and notice that the prompt shows up. Dismiss the prompt
- Publish a post again with sharing enabled and notice that no prompt shows up.
